x-pack/legacy/plugins/ml/public/overview/components/anomaly_detection_panel/anomaly_detection_panel.tsx

```diff
@@ -0,0 +1,209 @@
+  latest_timestamp: number;
+  max_anomaly_score: number | null;
+}
+
+export interface Job {
+  [key: string]: any;
+}
+
+export type Jobs = Job[];
+
+interface MaxScoresByGroup {
+  [key: string]: {
+    maxScore: number;
+    index?: number;
+  };
+}
+
+const createJobLink = '#/jobs/new_job/step/index_or_search';
+
+function getDefaultAnomalyScores(groups: Group[]): MaxScoresByGroup {
+  const anomalyScores: MaxScoresByGroup = {};
+  groups.forEach(group => {
+    anomalyScores[group.id] = { maxScore: 0 };
+  });
+
+  return anomalyScores;
+}
+
+export const AnomalyDetectionPanel: FC = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [groups, setGroups] = useState<Groups>({});
+  const [groupsCount, setGroupsCount] = useState<number>(0);
+  const [jobsList, setJobsList] = useState<Jobs>([]);
+  const [statsBarData, setStatsBarData] = useState<any>(undefined);
+  const [errorMessage, setErrorMessage] = useState<any>(undefined);
+
+  const loadJobs = async () => {
+    setIsLoading(true);
+
+    try {
+      const jobsResult: Jobs | any = await ml.jobs.jobsSummary([]);
+      const jobsSummaryList = jobsResult.map((job: any) => {
+        job.latestTimestampSortValue = job.latestTimestampMs || 0;
+        return job;
+      });
+      const jobsGroups = getGroupsFromJobs(jobsSummaryList);
+      const jobsWithTimerange = getJobsWithTimerange(jobsSummaryList);
+      const stats = getStatsBarData(jobsSummaryList);
+      setIsLoading(false);
+      setErrorMessage(undefined);
+      setStatsBarData(stats);
+      setGroupsCount(Object.values(jobsGroups).length);
+      setGroups(jobsGroups);
+      setJobsList(jobsWithTimerange);
+      loadMaxAnomalyScores(jobsGroups);
+    } catch (e) {
+      setErrorMessage(JSON.stringify(e));
+      setIsLoading(false);
+    }
+  };
+
+  const loadMaxAnomalyScores = async (groupsObject: Groups) => {
+    const groupsList: Group[] = Object.values(groupsObject);
+    const scores = getDefaultAnomalyScores(groupsList);
+
+    try {
+      const promises: any[] = [];
+      groupsList.forEach((group, i) => {
+        if (group.jobIds.length > 0) {
+          scores[group.id].index = i;
+          promises.push(
+            mlResultsService.getOverallBucketScores(
+              group.jobIds,
+              1,
+              group.earliest_timestamp,
+              group.latest_timestamp
+            )
+          );
+        }
+      });
+
+      const results = await Promise.all(promises);
+      const tempGroups = Object.assign({}, { ...groupsObject });
+      // Check results for each group's promise index and update state
+      Object.keys(scores).forEach(groupId => {
+        const resultsIndex = scores[groupId] && scores[groupId].index;
+        const promiseResult: { success: boolean; results: { [key: string]: number } } =
+          resultsIndex !== undefined && results[resultsIndex];
+        if (promiseResult.success === true && promiseResult.results !== undefined) {
```

Comment:
This throws an error if there are no ungrouped jobs, with `promiseResult` being `undefined`:

![image](https://user-images.githubusercontent.com/7405507/65158573-88cb0980-da2a-11e9-961d-91882ee33abf.png)

Reply:
Good catch, @peteharverson. This is occurring because we're passing the empty `jobIds` array. I've created a fix to remove `ungrouped` when there are no ungrouped job ids.